src/main/java/seedu/address/model/Model.java

```diff
@@ -19,14 +20,28 @@
     void resetData(ReadOnlyTaskBook newData);
 
     /** Returns the TaskBook */
-    ReadOnlyTaskBook getAddressBook();
+    ReadOnlyTaskBook getTaskBook();
 
     /** Deletes the given task. */
     void deleteTask(Task target) throws TaskNotFoundException;
 
     /** Adds the given task */
     void addTask(Task task);
 
+    //undo and redo
+    /** Undo prev action that modifies data
+     * @throws IllegalValueException **/
+    Command undo() throws IllegalValueException;
+
+    void recordState(Command command);
+
+    boolean hasUncommittedChanges();
+
+    /** redo previous undo
+     * @throws IllegalValueException **/
+    Command redo() throws IllegalValueException;
```

Comment:
Will this return the command of the commit before redoing, or the command of the commit that was redone? This needs to be stated.